tests/appsec/test_processor.py

```diff
@@ -18,13 +19,34 @@
 RULES_MISSING_PATH = os.path.join(ROOT_DIR, "nonexistent")
 
 
+class Config(object):
+    def __init__(self):
+        self.is_header_tracing_configured = False
+
+
 def _enable_appsec(tracer):
     tracer._appsec_enabled = True
     # Hack: need to pass an argument to configure so that the processors are recreated
     tracer.configure(api_version="v0.4")
     return tracer
 
 
+def test_transform_headers():
+    transformed = _transform_headers(
+        {
+            "hello": "world",
+            "Foo": "bar1",
+            "foo": "bar2",
+            "fOO": "bar3",
+            "BAR": "baz",
+            "COOKIE": "secret",
+        }
+    )
+    assert transformed["hello"] == "world"
```

Comment:
```suggestion
    assert set(transformed.keys()) == {"hello", "bar", "foo"}
    assert transformed["hello"] == "world"
```


an alternative that I really don't love... what you have now is better

```python
    transformed = _transform_headers(
        {
            "hello": "world",
            "Foo": "bar1",
            "foo": "bar2",
            "fOO": "bar3",
            "BAR": "baz",
            "COOKIE": "secret",
        }
    )
    expected = {"hello": "world", "foo": ["bar1", "bar2", "bar3"], "bar": "baz"}

    assert transformed.keys() == expected.keys()
    for key in transformed.keys():
        assert key in expected
        if isinstance(transformed[key], list):
            assert len(transformed[key]) == len(expected[key])
            assert set(transformed[key]) == set(expected[key])
        else:
            assert transformed[key] == expected[key]
```